### ISSUE TITLE
[AllPlay] Defined configuration values are ignored by the binding

### ISSUE THREAD

**Issue by dominicdesu** [openhab/openhab2-addons#2335]
## Expected Behavior
AllPlay binding should use the defined configuration values (such as "Rewind skip time")

## Current Behavior
Defined configuration values are ignored.

Continuing the discussion [from this PR](https://github.com/openhab/openhab2-addons/pull/1565), where @kaikreuzer suggested the following:

> The configs for bindings are sent to components with the name binding.<id> - so all you need to do is to adapt your component name from org.openhab.binding.allplay.internal.AllPlayHandlerFactory to binding.allplay.

**Comment by dominicdesu** [comment]
@kaikreuzer thanks for the hint! I changed the component name, unfortunately I am still not getting the config parameters... Also tried to set configuration-policy="optional" (same as in the sonos binding) but no effect.

Any idea how I could debug this problem?
I already found out that in BundlecontextImpl.registerService(), the properties are already missing (only component.name and component.id are available). Unfortunately I cannot find the place where the properties should actually be set (seems to be deep down in some library?)

Also, when installing the sonos binding, the properties contain more values, such as service.pid and felix.fileinstall.filename. Also sonos binding is adding a file /userdata/etc/binding.sonos.cfg, which I cannot see for the allplay binding.

This is my AllPlayHandlerFactory.xml:
```
<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" configuration-policy="optional" immediate="true" name="binding.allplay">
   <implementation class="org.openhab.binding.allplay.internal.AllPlayHandlerFactory"/>
   <service>
      <provide interface="org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory"/>
   </service>
   <reference bind="setAudioHTTPServer" cardinality="1..1" interface="org.eclipse.smarthome.core.audio.AudioHTTPServer" name="AudioHTTPServer" policy="static" unbind="unsetAudioHTTPServer"/>
</scr:component>
```

**Comment by kaikreuzer** [comment]
I just tested it and it works all as expected. The correct name is indeed `name="binding.allplay"`, just as you have it in your xml. I have now configured some values for the binding in the Paper UI and directly when I hit the save button there, I hit my breakpoint:

<img width="1184" alt="screen shot 2017-06-09 at 23 17 36" src="https://user-images.githubusercontent.com/3244965/26994903-30d8ec3c-4d6a-11e7-83df-2fa6d08aeaef.png">

As you can see, the properties contain all the configuration that is done in the Paper UI.

**Comment by dominicdesu** [comment]
@kaikreuzer, thanks for investing the time and trying this out! I'm very curious why it isn't working for me. :-/ Will try again with a fresh install as soon as I have some time.

**Comment by dominicdesu** [comment]
Thanks @kaikreuzer, it finally worked by only changing the name in the xml. Don't know what I did wrong some weeks ago :-/

**Comment by dominicdesu** [comment]
@kaikreuzer I think I know now what went wrong and it is a little confusing:
* If I add a new parameter (with a default value) in the binding.xml, the parameter does not appear in the component properties initially.
* Only after I've opened the binding configuration in PaperUI and press save (without changing any values), the component properties start to include my new parameters.

Is this on purpose? I was expecting that I get the default value if the user doesn't change it?

**Comment by kaikreuzer** [comment]
Yes, this is on purpose, at least partially. The main idea of the "default" information in the config description is to tell the UI that it can suggest something as a default value. In general, a parameter with a default value should NOT be marked required, thus it is fully ok to not provide this value in the configuration.
That the UI fills this parameter although nothing has changed, is indeed a bit surprising.

**Comment by dominicdesu** [comment]
Okay if I understand it correctly, I need to use an "internal" default value (besides the one defined in binding.xml), in case the configuration parameter is not yet set. In this case I should update my PR 😃 
Thanks for the explanation!

**Comment by dominicdesu** [comment]
@kaikreuzer this ticket has been fixed by #2501
